src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -589,12 +590,12 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
       BuildOptions defaultBuildOptions)
       throws DependencyEvaluationException, ConfiguredTargetFunctionException,
           AspectCreationException, InterruptedException {
-    // Create the map from attributes to set of (target, configuration) pairs.
-    OrderedSetMultimap<DependencyKind, Dependency> depValueNames;
+    // Create the map from attributes to set of (target, transition) pairs.
+    OrderedSetMultimap<DependencyKind, DependencyKey> initialDependencies;
```

Comment:
Open question (but not for this PR): I wonder if we can even fold DependencyKind into DependencyKey?

Reply:
I don't think it makes sense: the DependencyKind is effectively an enum: it distinguishes different types of dependency (toolchain, visibility, output_group, or different attributes). Whereas the DependencyKey is a single dependency: it's a label, a config transition, and maybe a bit more data, but it doesn't care what the key was (except in a few cases where we are trying to optimize).

I have more work on ConfigurationResolver, so we can revisit this when those changes are ready for review.